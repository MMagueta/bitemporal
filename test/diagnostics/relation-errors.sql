\set QUIET 'on'
\pset footer 'off'

drop schema if exists re cascade;
drop schema if exists re_history cascade;
create schema re;
create schema re_history;

\echo 'bitemporal.relation_errors - [ok]'
\pset title 're.ok'
create table re.ok
    ( id                int not null
    , id2               int not null
    , value             int not null
    , value2            int not null
    , valid_period      int4range not null
    , system_period     int4range not null
    , primary key(id, id2, valid_period)
    , exclude using gist (id with =, id2 with =, valid_period with &&)
    , exclude using gist (id with =, id2 with =, value with =, value2 with =, valid_period with -|-) );
create table re_history.ok( like re.ok );
select * from bitemporal.relation_errors('re.ok');

\echo 'bitemporal.relation_errors - [not ok]'
\pset title 're.not_ok'
create table re.not_ok( id int );
select * from bitemporal.relation_errors('re.not_ok');
