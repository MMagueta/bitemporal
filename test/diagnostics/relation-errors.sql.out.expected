             re.ok
 namespace | relation | message 
-----------+----------+---------

                              re.not_ok
 namespace | relation |                   message                    
-----------+----------+----------------------------------------------
 re        | not_ok   | Missing attribute "valid_period"
 re        | not_ok   | Missing attribute "system_period"
 re        | not_ok   | Missing primary key
 re        | not_ok   | Missing exclude overlap constraint
 re        | not_ok   | Missing exclude adjacency constraint
 re        | not_ok   | Missing history relation "re_history.not_ok"

